### PR TITLE
Cmd pushall fixes

### PR DESCRIPTION
**Pull request by venkatvghub**
- kubernetes api client fixes 
- fixes aws boto region for handling non default regions in dynamodb stream client
- dynamodb stream fix to handle shard iteration
- fixes exception/error handling for dynamodb responses

### CODE REVIEW

File: kubestash/__init__.py [hunk 1]
```diff
@@ -294,14 +298,20 @@ def maybe_reverse_dns_subdomain(args, string):
...
     """
-    api_client = kubernetes.client.ApiClient(configuration=kubernetes.client.configuration)
-    return kubernetes.client.CoreV1Api(api_client=api_client)
+    if args.in_cluster == False:
+        kubernetes.config.load_kube_config()
+    else:
+        kubernetes.config.load_incluster_config()
+    api_client = kubernetes.client.CoreV1Api()
+    return api_client
+    #api_client = kubernetes.client.ApiClient(configuration=kubernetes.client.configuration)
```
Comment (captn3m0) [inline kubestash/__init__.py]: remove commented code

File: kubestash/__init__.py [hunk 2]
```diff
@@ -584,27 +594,35 @@ def get_stream_client(args):
...
-        sys.exit(1)
+        if not response['StreamDescription']:
+            print("fatal: no StreamDescription found for DynamoDB stream '{arn}'.".format(arn=arn))
+            sys.exit(1)
 
-    shard_id = response['StreamDescription']['Shards'][0]['ShardId']
+        if not response['StreamDescription']['Shards']:
+            print("fatal: no shards found for DynamoDB stream '{arn}'.".format(arn=arn))
+            sys.exit(1)
+        try:
```
Comment (captn3m0) [inline kubestash/__init__.py]: Wondering if the pod will restart properly after this. (What do we keep as a healthcheck?)

File: kubestash/__init__.py [hunk 3]
```diff
@@ -661,25 +679,34 @@ def cmd_daemonall(args):
...
+
+            if len(records) > 0:
+                for record in records:
+                    try:
+                        key = record['dynamodb']['Keys']['name']['S']
+                        if args.verbose:
+                            print("detected DynamoDB changes, running push command...")
+                        argscopy = copy.copy(args)
+                        argscopy.secretname = key
+                        cmd_pushall(argscopy)
```
Comment (captn3m0) [inline kubestash/__init__.py]: Why are we still doing a pushall? I thought the idea was to run a selective patch on the secret.ENV for every change.

Comment (venkatvghub) [reply kubestash/__init__.py]: @captn3m0 need some clarity on this. Lets sync up when am back from indore

**Pull request merged**